From 278ca3f3738f6e69b81b6aa776ca545f24cd942c Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 14 Aug 2019 16:50:12 +0200
Subject: [PATCH] tqma8mx: ft_board_setup improvements for older CPU revisions

- not delete cpu operating-points (implemented in base soc dt)
- delete experimental 1000 mV fixed regulator (overdrive without voltage scaling test)

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 board/tqc/tqma8mx/tqma8mx.c | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/board/tqc/tqma8mx/tqma8mx.c b/board/tqc/tqma8mx/tqma8mx.c
index 260a8308ed..2feeea7d0e 100644
--- a/board/tqc/tqma8mx/tqma8mx.c
+++ b/board/tqc/tqma8mx/tqma8mx.c
@@ -93,15 +93,14 @@ int ft_board_setup(void *blob, bd_t *bd)
 		if (nodeoff < 0) {
 			printf("Unable to find /cpus/cpu@0\n");
 		} else {
-			rc = fdt_delprop(blob, nodeoff,
-					  "operating-points");
-			if (rc < 0)
-				printf("Unable to delete operating-points, err=%s\n",
-					fdt_strerror(rc));
 			rc = fdt_delprop(blob, nodeoff, "dc-supply");
 			if (rc < 0)
 				printf("Unable to delete prop dc-supply, err=%s\n",
 					fdt_strerror(rc));
+			rc = fdt_delprop(blob, nodeoff, "arm-supply");
+			if (rc < 0)
+				printf("Unable to delete prop arm-supply, err=%s\n",
+					fdt_strerror(rc));
 		}
 
 		nodeoff = fdt_path_offset(blob, "/tqma8mx-vdd-arm");
@@ -113,6 +112,16 @@ int ft_board_setup(void *blob, bd_t *bd)
 				printf("Unable to delete /tqma8mx-vdd-arm, err=%s\n",
 					fdt_strerror(rc));
 		}
+
+		nodeoff = fdt_path_offset(blob, "/reg_tqma8mx_overdrive");
+		if (nodeoff < 0) {
+			printf("Unable to find /reg_tqma8mx_overdrive\n");
+		} else {
+			rc = fdt_del_node(blob, nodeoff);
+			if (rc < 0)
+				printf("Unable to delete /reg_tqma8mx_overdrive, err=%s\n",
+					fdt_strerror(rc));
+		}
 	}
 
 	return tqc_bb_ft_board_setup(blob, bd);
